tests/integration/defs/perf/test_perf.py

```diff
@@ -375,6 +415,284 @@ class PerfTestMetric(NamedTuple):
     cmd_idx: int
 
 
+class ServerConfig:
+    """
+    Configurations of trtllm-server.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        model_name: str,
+        tp: int,
+        ep: int,
+        max_num_tokens: int,
+        attention_backend: str,
+        max_batch_size: int,
+        pp: int = 1,
+        enable_chunked_prefill: bool = False,
+        disable_overlap_scheduler: bool = False,
+        moe_backend: str = "",
+        moe_max_num_tokens: str = "",
+        enable_attention_dp: bool = False,
+        attention_dp_balance: bool = False,
+        batching_wait_iters: int = 10,
+        timeout_iters: int = 50,
+        kv_cache_dtype: str = "fp8",
+        enable_block_reuse: bool = False,
+        free_gpu_memory_fraction: float = 0.8,
+        enable_padding: bool = True,
+    ):
+        self.name = name
+        self.model_name = model_name
+        self.tp = tp
+        self.ep = ep
+        self.pp = pp
+        self.max_num_tokens = max_num_tokens
+        self.enable_chunked_prefill = enable_chunked_prefill
+        self.disable_overlap_scheduler = disable_overlap_scheduler
+        self.attention_backend = attention_backend
+        self.moe_backend = moe_backend
+        self.moe_max_num_tokens = moe_max_num_tokens
+        self.enable_attention_dp = enable_attention_dp
+        self.attention_dp_balance = attention_dp_balance
+        self.batching_wait_iters = batching_wait_iters
+        self.timeout_iters = timeout_iters
+        self.kv_cache_dtype = kv_cache_dtype
+        self.enable_block_reuse = enable_block_reuse
+        self.free_gpu_memory_fraction = free_gpu_memory_fraction
+        self.max_batch_size = max_batch_size
+        self.enable_padding = enable_padding
+
+        self.model_path = ""
+
+    def to_cmd(self, working_dir: str) -> List[str]:
+        model_dir = get_model_dir(self.model_name)
+        self.model_path = model_dir if os.path.exists(
+            model_dir) else self.model_name
+        config_path = os.path.join(working_dir,
+                                   f"extra-llm-api-config.{self.name}.yml")
+        return [
+            "trtllm-serve", self.model_path, "--host", "localhost", "--port",
+            "8000", "--backend", "pytorch", "--extra_llm_api_options",
+            config_path
+        ]
+
+    def generate_extra_llm_api_config(self) -> str:
+        """Generate extra-llm-api-config.yml content"""
+        config_lines = [
+            f"tensor_parallel_size: {self.tp}",
+            f"moe_expert_parallel_size: {self.ep}",
+            f"pipeline_parallel_size: {self.pp}",
+            f"max_num_tokens: {self.max_num_tokens}",
+            f"enable_attention_dp: {str(self.enable_attention_dp).lower()}",
+            f"disable_overlap_scheduler: {str(self.disable_overlap_scheduler).lower()}",
+            "stream_interval: 10",
```

Comment:
I sugges to store this in the  test config, to let user override it, instead of hard coding it to 10.